types/passport-instagram/index.d.ts

```diff
@@ -0,0 +1,55 @@
+// Type definitions for passport-instagram 1.0.0
+// Project: https://github.com/jaredhanson/passport-instagram
+// Definitions by: Sagar Jain <https://github.com/sagar7993>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+
+/// <reference types="passport"/>
+
+
+
+import passport = require('passport');
+import express = require('express');
+
+interface Profile extends passport.Profile {
+    gender: string;
+    username: string;
+
+    _raw: string;
+    _json: any;
+    _accessLevel: string;
+}
+
+interface IStrategyOptionBase {
+    consumerKey: string;
+    consumerSecret: string;
+    callbackURL: string;
+
+    includeEmail?: true;
+
+    reguestTokenURL?: string;
+    accessTokenURL?: string;
+    userAuthorizationURL?: string;
+    sessionKey?: string;
+
+    userProfileURL?: string;
+    skipExtendedUserProfile?: boolean;
+}
+
+interface IStrategyOption extends IStrategyOptionBase {
+    passReqToCallback?: false;
+}
+
+interface IStrategyOptionWithRequest extends IStrategyOptionBase {
+    passReqToCallback: true;
+}
+
+declare class Strategy extends passport.Strategy {
```

Comment:
I recommend `export class` instead of `declare class` for clarity.

Reply:
Done